low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,500 @@
+    SUCCESS = "Success"
+    FAILED = "Failed"
+
+    def get_processing_status(req_id):
+        """
+        Returns "in-progress" if any request is being processed.
+        Otherwise returns "Success" or "Failed (with reason)" status.
+        """
+        if not req_id:
+            raise SupportBundleError(errno.EINVAL, "Invalid request ID.")
+        status_list = req_register.get(["%s>status" % req_id])
+        status = status_list[0] if status_list else None
+
+        if not status:
+            raise SupportBundleError(
+                errno.EINVAL, "Request ID '%s' not found." % req_id)
+        else:
+            # Set failed status to stale request ID
+            expiry_sec = 1800
+            last_reboot = int(psutil.boot_time())
+            # Set request is expired if processing time exceeds
+            system_time = time.strptime(
+                req_register.get(["%s>time" % req_id])[0],
+                '%Y-%m-%d %H:%M:%S')
+            req_start_time = int(time.mktime(system_time))
+            current_time = int(time.time())
+            is_req_expired = (current_time - req_start_time) > expiry_sec
+            if is_req_expired or (last_reboot > req_start_time and
+                                  status is BundleRequestHandler.INPROGRESS):
+                # Set request state as failed
+                BundleRequestHandler._update_request_status(
+                    req_id, "Failed - request is expired.")
+            status = req_register.get(["%s>status" % req_id])[0]
+
+        return status
+
+    def _update_request_status(req_id, status):
+        """Updates processed request information."""
+        req_info = req_register.get(["%s" % req_id])[0]
+        req_info.update({
+            "status": status,
+            "time": datetime.strftime(
+                datetime.now(), '%Y-%m-%d %H:%M:%S')
+        })
+        req_register.set(["%s" % req_id], [req_info])
+
+    @staticmethod
+    def _add_bundle_request(req_id, url):
+        """Updates new request information."""
+        req_info = {
+            "status": BundleRequestHandler.INPROGRESS,
+            "url": url,
+            "time": datetime.strftime(
+                datetime.now(), '%Y-%m-%d %H:%M:%S')
+        }
+        req_register.set(["%s" % req_id], [req_info])
+
+
+class SupportBundle(object):
+    """SSPL support bundle class"""
+
+    def __init__(self):
+
+        self.SYS_INFORMATION = "SYSTEM_INFORMATION"
+        self.IEM_SENSOR = "IEMSENSOR"
+        self._tmp_dir = '/tmp/cortx'
+        self._default_path = '%s/support_bundle/' % self._tmp_dir
+        self.tmp_bundle_path = "/tmp/support_bundle/"
+        self.sspl_log_default = f"/var/log/{sspl_const.PRODUCT_FAMILY}/sspl"
+        self.iem_log_default = f"/var/log/{sspl_const.PRODUCT_FAMILY}/iem"
+        self.sspl_conf_dir = f"{sspl_const.SSPL_SB_TMP}/sspl_conf/"
+        self.ipmi_sel_data = f"{sspl_const.SSPL_SB_TMP}/ipmi_sel_data_{str(int(time.time()))}.txt"
```

Comment:
please make a limit of max 80 characters in a line.
split line when the characters count goes above 80.
This will be applicable across all files in this PR.

Reply:
I have restricted most of the lines to <=80 chars. Although some lines are upto 90 chars as reducing them will either break that line or will reduce the readability.

Reply:
Okay, Done